sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -652,11 +652,18 @@ class Analyzer(
             case s: Star => s.expand(child, resolver)
             case o => o :: Nil
           })
-        case c: CreateStruct if containsStar(c.children) =>
-          c.copy(children = c.children.flatMap {
-            case s: Star => s.expand(child, resolver)
-            case o => o :: Nil
-          })
+        case c: CreateNamedStruct if containsStar(c.valueExpressions) =>
+          val newChildren = for {
+            kv @ Seq(k, v) <- c.children.grouped(2)
+            expanded = v match {
```

Comment:
it's pretty weird to put a pattern match inside a for clause, can we use a flatMap instead?